lib/ansible/modules/cloud/amazon/ec2_ami.py

```diff
@@ -677,7 +677,6 @@ def main():
         argument_spec=argument_spec,
         required_if=[
             ['state', 'absent', ['image_id']],
-            ['state', 'present', ['name']],
```

Comment:
@willthames should there be a required_one_of=[['name', 'image_id']] for this? Now it may try to create an AMI without a name if no image_id is provided.

Reply:
Also disagree that name should default to an empty string since that isn't documented and it isn't used at all in the update image function. Defaulting to an empty string was probably just done to allow this check to be bypassed which was no longer necessary once this line was removed. I've opened https://github.com/ansible/ansible/pull/38972 which also has tests to make sure backward compatibility isn't broken again. If that gets merged I'll cherry pick that commit onto maishsk's pull request for backporting (thanks for opening that and working on this, @maishsk!).